share/smack/prelude.py

```diff
@@ -0,0 +1,290 @@
+import itertools
+
+class Attribute:
+  def __init__(self, attribute, attribute_arg = None):
```

Comment:
This seems vague and redundant. How about `name` and `arguments` rather than `attribute` and `attribute_arg`. (Note that Boogie attribute arguments are generally a comma separated list, so this should probably be a list too.)